frontend/src/components/Appbar.tsx

```diff
@@ -14,64 +15,97 @@ const Appbar = ({ skipAuthCheck = false, pageActions, hideWriteAction = false }:
   const { pathname } = useLocation();
   const isUserLoggedIn = localStorage.getItem('token');
 
+  const [isDarkMode, setIsDarkMode] = useState(() => {
+    return localStorage.getItem('theme') === 'dark';
+  });
+
+  const toggleDarkMode = () => {
+    if (isDarkMode) {
+      document.documentElement.classList.remove('dark');
+      localStorage.removeItem('theme');
+    } else {
+      document.documentElement.classList.add('dark');
+      localStorage.setItem('theme', 'dark');
+    }
+    setIsDarkMode(!isDarkMode);
+  };
+
+  useEffect(() => {
+    if (isDarkMode) {
+      document.documentElement.classList.add('dark');
+    } else {
+      document.documentElement.classList.remove('dark');
+    }
+  }, [isDarkMode]);
+  
+
```

Comment:
Create a custom hook for this snippet and call the custom here instead of coupling the theme logic with the component.